cloudinstall/charms/__init__.py

```diff
@@ -369,10 +379,16 @@ def watch_relations_async(self):
         self.watch_relations()
 
     def watch_relations(self):
+        """
+        Handle charm relations
+        """
+        for charm in self.deployed_charms:
+            self.charm_relations_q.put(charm)
         log.debug("Starting charm relations watcher.")
         while not self.charm_relations_q.empty():
             try:
                 charm = self.charm_relations_q.get()
+                charm.related = self.filter_valid_relations()
```

Comment:
filter_valid_relations is called here without an arg. we should probably have a test for watch_relations, which would have caught this